python/dolfinx/io/vtkhdf.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2024 Chris Richardson
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+
+import numpy as np
+
+import basix
+import ufl
+from dolfinx.cpp.io import read_vtkhdf_mesh_float32, read_vtkhdf_mesh_float64, write_vtkhdf_mesh
+from dolfinx.mesh import Mesh
+
+
+def read_mesh(comm, filename, dtype=np.float64):
```

Comment:
```suggestion
import typing
from pathlib import Path

from mpi4py import MPI as _MPI

import numpy as np

import basix
import ufl
from dolfinx.cpp.io import read_vtkhdf_mesh_float32, read_vtkhdf_mesh_float64, write_vtkhdf_mesh
from dolfinx.mesh import Mesh


def read_mesh(comm: _MPI.Comm, filename: typing.Union[str, Path], dtype=np.float64):
    """Read mesh data from file.
    
    Args:
        comm: MPI-communicator to distribute mesh over.
        filename: Path to input file.
        dtype: Data-type of mesh geometry.
    """
```
Can we not determine the dtype when looking for a specific variable with HDF5?